contrib/ros/envs/sawyer_env.py

```diff
@@ -0,0 +1,346 @@
+        """
+        self.goal = copy.deepcopy(self.sample_goal())
+
+        if self._simulated:
+            self._gazebosrv.set_model_pose(
+                model_name='target',
+                new_pose=Pose(
+                    position=Point(
+                        x=self.goal[0], y=self.goal[1], z=self.goal[2])),
+                reference_frame='world')
+            self._reset_sim()
+        else:
+            # TODO(gh/3: Finish the reset function for real robot)
+            pass
+        obs = self._get_obs()
+        initial_observation = obs['observation']
+
+        return initial_observation.copy()
+
+    def _reset_sim(self):
+        """
+        reset the simulation
+        """
+        self._robot.move_to_start_position()
+
+        # Randomize start position of object
+        if self.has_object:
+            object_xypos = self.initial_gripper_pos[:2]
+            while np.linalg.norm(object_xypos -
+                                 self.initial_gripper_pos[:2]) < 0.1:
+                object_xypos = self.initial_gripper_pos[:
+                                                        2] + self.np_random_generator.uniform(
+                                                            -self.obj_range,
+                                                            self.obj_range,
+                                                            size=2)
+            self._gazebosrv.set_model_pose(
+                model_name='block',
+                new_pose=Pose(
+                    position=Point(
+                        x=object_xypos[0],
+                        y=object_xypos[1],
+                        z=self._initial_model_pos['object0'][2])),
+                reference_frame='world')
+
+    @property
+    def action_space(self):
+        return self._robot.action_space
+
+    @property
+    def observation_space(self):
+        """
+        Returns a Space object
+        """
+        return Box(-np.inf, np.inf, shape=self._get_obs()['observation'].shape)
+
+    # ================================================
+    # Functions that gazebo env asks to implement
+    # ================================================
+    def initial_setup(self, initial_model_pos):
+        # Extract information for sampling goals
+        gripper_pose = self._robot.gripper_pose
+        self.initial_gripper_pos = np.array(gripper_pose['position'])
+        self.initial_gripper_ori = np.array(gripper_pose['orientation'])
+        if self.has_object:
+            self.height_offset = initial_model_pos['object0'][2]
+
+        self._robot.move_to_start_position()
+
+        if self._simulated:
+            # Generate the world
+            # Load the model
+            self._gazebosrv.load_gazebo_sdf_model(
+                model_name='cafe_table',
+                model_pose=Pose(
+                    position=Point(initial_model_pos['table0'][
```

Comment:
please don't include huge expressions in kwargs. Make temporary variables for all these.

Reply:
Okay